### DISCUSSION TITLE
Why is does this excludePackagePattern not work?

### DISCUSSION THREAD

**Discussion by helpermethod** [renovatebot/renovate#18821] [Request Help] [answered]
Hi!

For some reason, the excludePackagePattern `"^io.getunleash:unleash-client-java:.*"` does not work

```
    {
      "matchPackagePatterns": [
        ".*"
      ],
      "excludePackagePatterns": [
        "^io.getunleash:unleash-client-java:6.0.2:.*"
      ]
```

but `"^io.getunleash:.*"` does

```
    {
      "matchPackagePatterns": [
        ".*"
      ],
      "excludePackagePatterns": [
        "^io.getunleash:.*"
      ]
```

Im using Gradle, where the dependencies' coordinate looks like this

    implementation("io.getunleash:unleash-client-java:6.0.2")
    
As `excludePackagePatterns` accepts a regex, I would expect both pattern to work.

**Comment by viceice** [comment, marked as answer]
check the `packagefiles with updates` debug message. there you find the names of each package.

your regex can't match `implementation("com.fasterxml.jackson.module:jackson-module-kotlin")` because of the `:` before `.*`

**Comment by helpermethod** [comment]
Sorry, messed up the example. There coordinates are actually

    implementation("io.getunleash:unleash-client-java:6.0.2")

And I've added the correct patterns.

**Reply by viceice** [reply]
you can't match the version. you can only match the string before.

**Reply by viceice** [reply]
i also told you to check the debug logs. there you can find the package name you can match against.

**Reply by helpermethod** [reply]
Ah sorry, that makes sense!
